org.mixedrealitytoolkit.input/Interactors/Gaze/TrackedPoseDriverWithFallback.cs

```diff
@@ -0,0 +1,116 @@
+        /// action has no data. If not specified, this will fallback to the device's tracking state that drives the position
+        /// or rotation action. Must be a IntegerControl Control.
+        /// </summary>
+        public InputActionProperty FallbackTrackingStateAction => fallbackTrackingStateAction;
+
+        #endregion Fallback action values
+
+        #region TrackedPoseDriver Overrides 
+        /// <inheritdoc />
+        protected override void PerformUpdate()
+        {
+            base.PerformUpdate();
+
+            if (trackingStateInput.action == null)
+            {
+                Debug.LogWarning("TrackedPoseDriverWithFallback.trackingStateInput.action is null, no fallback will be used.");
+                return;
+            }
+
+            var hasPositionAction = positionAction != null;
+            var hasPositionFallbackAction = fallbackPositionAction != null;
+
+            var hasRotationAction = rotationAction != null;
+            var hasRotationFallbackAction = fallbackRotationAction != null;
+
+            InputTrackingState inputTrackingState = (InputTrackingState)trackingStateInput.action.ReadValue<int>();
+
+            // If default InputTrackingState does not have position and rotation data, use fallback if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Position) && !inputTrackingState.HasFlag(InputTrackingState.Rotation) && FallbackTrackingStateAction.action != null)
+            {
+                inputTrackingState = (InputTrackingState)FallbackTrackingStateAction.action.ReadValue<int>();
+            }
+
+            bool neededToGetFallbackData = false;
+            Vector3 position = Vector3.zero;
+            Quaternion rotation = Quaternion.identity;
+
+            // If no position data then use the data from the fallback action if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Position) && hasPositionFallbackAction)
+            {
+                neededToGetFallbackData = true;
+                position = fallbackPositionAction.action.ReadValue<Vector3>();
+            }
+            else if (hasPositionAction)
+            {
+                position = positionAction.ReadValue<Vector3>(); //get a hold on the position data in case orientation data is obtained from fallback action
+            }
+
+            // If no rotation data then use the data from the fallback action if it exists
+            if (!inputTrackingState.HasFlag(InputTrackingState.Rotation) && hasRotationFallbackAction)
+            {
+                neededToGetFallbackData = true;
+                rotation = fallbackRotationAction.action.ReadValue<Quaternion>();
+            }
+            else if (hasRotationAction)
+            {
+                rotation = rotationAction.ReadValue<Quaternion>(); //get a hold on the rotation data in case position data was obtained from fallback action
+            }
```

Comment:
I don't like handling the "default" case here.  We should allow the "base" functionality to work as is.

Take a look at the base.PerformUpdate() and observe how it is setting `transform.localPosition` and `transform.localRotation`.

Instead have the starting rotation equal `transform.localRotation`

`Quaternion rotation = transform.localRotation`

Then remove this `else` case.

Reply:
resolving https://github.com/MixedRealityToolkit/MixedRealityToolkit-Unity/pull/746/commits/758fcaff7754d6479e2a54e9636a20708e0ae828